openslides_backend/action/action_worker.py

```diff
@@ -0,0 +1,288 @@
+from gunicorn.workers.gthread import ThreadWorker
+
+from openslides_backend.shared.patterns import fqid_from_collection_and_id
+
+from ..services.datastore.interface import DatastoreService
+from ..shared.exceptions import ActionException, DatastoreException
+from ..shared.interfaces.event import Event, EventType
+from ..shared.interfaces.logging import LoggingModule
+from ..shared.interfaces.write_request import WriteRequest
+from .action_handler import ActionHandler
+from .util.typing import ActionsResponse, Payload
+
+THREAD_WATCH_TIMEOUT = 1.0
+
+
+def handle_action_in_worker_thread(
+    payload: Payload,
+    user_id: int,
+    is_atomic: bool,
+    handler: ActionHandler,
+) -> ActionsResponse:
+    starttime = round(time())
+    curr_thread = threading.current_thread()
+    logger = handler.logging.getLogger(__name__)
+    lock = threading.Lock()
+    try:
+        action_names = ",".join(elem.get("action", "") for elem in payload)
+    except Exception:
+        action_names = "Cannot be determined"
+    action_worker_writing = ActionWorkerWriting(
+        user_id, starttime, handler.logging, action_names, handler.services.datastore()
+    )
+    action_worker_thread = ActionWorker(
+        payload,
+        user_id,
+        is_atomic,
+        handler,
+        lock,
+        cast(int, curr_thread.native_id),
+        action_worker_writing,
+    )
+    action_worker_thread.start()
+    while not action_worker_thread.started:
+        sleep(0.001)  # The action_worker_thread should gain the lock and NOT this one
+    if lock.acquire(timeout=THREAD_WATCH_TIMEOUT):
+        if hasattr(action_worker_thread, "exception"):
+            logger.debug("Action request finished with exception within timeout.")
+            lock.release()
+            raise action_worker_thread.exception
+        if hasattr(action_worker_thread, "response"):
+            logger.debug("Action request finished successfully within timeout.")
+            lock.release()
+            return action_worker_thread.response
+        lock.release()
+        raise ActionException(
+            "Action request ended with unknown reason, probably an unexpected timeout!"
+        )
+
+    message = action_worker_writing.initial_action_worker_write()
+    return ActionsResponse(
+        status_code=HTTPStatus.ACCEPTED.value,
+        success=False,
+        message=message,
+        results=[
+            [
+                {
+                    "fqid": action_worker_writing.fqid,
+                    "name": action_worker_writing.action_names,
+                    "written": action_worker_writing.written,
+                }
+            ]
+        ],
+    )
+
+
+class ActionWorkerWriting(object):
+    def __init__(
+        self,
+        user_id: int,
+        starttime: int,
+        logging: LoggingModule,
+        action_names: str,
+        datastore: DatastoreService,
+    ) -> None:
+        self.user_id = user_id
+        self.starttime = starttime
```

Comment:
Can't you just set the time here instead of using a parameter? I don't see how these milliseconds will change anything and saving parameters (especially if there are this many) is always better.

But If you insist on it, please use snake case (`start_time`).

Reply:
done and done